File: src/main/java/com/example/today_i_baekjoon/domain/Commit.java
package com.example.today_i_baekjoon.domain;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDate;

@Getter
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Entity
@Table(name = "commits")
public class Commit {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	@Column(nullable = false)
	private LocalDate commitDate;

	@Column(nullable = false, unique = true)
	private String commitUrl;
	
	@Column
	private String problemRank;
	
	@Column(nullable = false)
	private String problemTitle;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "user_id")
	private User user;
}

File: back/src/main/kotlin/com/gidskql6671/today_i_baekjoon/domain/Commit.kt
package com.gidskql6671.today_i_baekjoon.domain

import jakarta.persistence.*
import java.time.LocalDate

@Entity
@Table(name = "commits")
class Commit(
    @Column(nullable = false)
    var commitDate: LocalDate,

    @Column(nullable = false, unique = true)
    var commitUrl: String,

    var problemRank: String,

    @Column(nullable = false)
    var problemTitle: String,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    var user: User
) {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null
}